Jip-37 CORS 설정, OAuth 및 일반 로그인 구현

tmam444 opened the pull request:
### 개요
- CORS 설정
- OAuth 및 일반 로그인 API

### 작업 사항
[커밋1](https://github.com/jiphyeonjeon-42/backend-express/commit/614cb4072c648bbe247af22b65ce6695721ba730)
- app.ts에 cors 설정
- package.json에 library 추가 (cors, bcrypt)

[커밋2](https://github.com/jiphyeonjeon-42/backend-express/commit/6af243c505025a696c21a2ef11e22c1ca7ca72e5)
- OAuth 로그인 설정 (/auth/oauth)
- 일반 로그인 설정 (/auth/login)
- 현재 로그인 계정 정보 (/auth/me)
- JWT 설정 (auth.jwt.ts) 
   => 설명 필요할거 같아서 주석 작성했음
- 토큰 검증 (auth.strategy.ts)
- 에러 처리 interface 추가 (auth.interface.ts)

### 변경점
- localhost:4242에서 API 받을 수 있게 CORS 설정
- auth.validate.ts의 내용을 auth.jwt.ts로 옮김
    => JWT 관련된 함수라서 합쳐놓았음

### 목적
- 개발 및 코드 최적화

### 스크린샷 (optional)

DiceMono commented:
프론트와 협의하여 기존 메인페이지에서 사용하던 검색 api url 을 수정하였습니다.

min-kyung reviewed (approved):
book url 수정 및 book search 코드 확인했습니다.

tmam444 committed:
JIP-24 ADD : Auth API 명세 초안 작성(swagger)
- login, logout, register swagger 초안 명세 작성!

tmam444 committed:
Backup

tmam444 committed:
Merge branch 'develop' into JIP-37

tmam444 committed:
JIP-37 CORS 설정 및 package.json에 library 추가
    - app.ts에 cors 설정 (localhost:4242, 42library.kr)
    - package.json에 cors 라이브러리, bcrypt 라이브러리 추가했슴..

tmam444 committed:
JIP-37 [Auth] OAuth, 일반 로그인, 로그인 정보 API
    - OAuth 로그인 설정 (/auth/oauth)
    - 일반 로그인 설정 (/auth/login)
    - 현재 로그인 계정 정보 (/auth/me)
    - JWT 설정 (auth.jwt.ts)
    - 토큰 검증 (auth.strategy.ts)

tmam444 commented:
가장 밑에 두개 커밋만 보시면 됩니닷ㅋㅋ

jimin52 commented:
정확하게 알지는 못하지만 3000번 포트에서 clientUrl 받은 이유가 .env 에서 clientUrl 받기 위한 거 아니었나요? 백엔드 서버만 돌리면서 clientUrl은 받아올 수 있는건가요?

kth2624 reviewed (approved):
고생하셨습니다.

YeonSeong-Lee commented:
database에 유저테이블 변경하는 부분에서
 스키마구조가 전반적으로 바뀌면서 처음 user데이터를 생성할 때 현재 DB명세에 맞게 바꾸어서 넣어주기때문 필요없을 듯 합니다.
  [데이터스키마수정 및 데이터추가  PR](https://github.com/jiphyeonjeon-42/backend-express/pull/39)

tmam444 commented:
> 정확하게 알지는 못하지만 3000번 포트에서 clientUrl 받은 이유가 .env 에서 clientUrl 받기 위한 거 아니었나요? 백엔드 서버만 돌리면서 clientUrl은 받아올 수 있는건가요?

확인 완료!!!

Commit:
JIP-37 CHORE: 사용하지 않을 SQL 파일 삭제

tmam444 commented:
> database에 유저테이블 변경하는 부분에서
> 스키마구조가 전반적으로 바뀌면서 처음 user데이터를 생성할 때 현재 DB명세에 맞게 바꾸어서 넣어주기때문 필요없을 듯 합니다.
> [데이터스키마수정 및 데이터추가 PR](https://github.com/jiphyeonjeon-42/backend-express/pull/39)

확인했습니다!! SQL 파일 삭제 했습니다!!
